Tests/test-apply.ts

```diff
@@ -12,6 +12,7 @@ tr.setInput('kubectlBinary', './Tests/kubectl');
 tr.setInput('k8sService', 'k8sendpoint');
 tr.setInput('system.debug', 'true');
 
+process.env['SYSTEM_DEFAULTWORKINGDIRECTORY'] = '/opt/vsts/work/r1/a'
```

Comment:
You should not hardcode the directory `opt/vsts/work/r1/a`

Reply:
Thank you, good review. I'll fix that.

Reply:
Hi @dtzar 

I've done some research to possible refactoring of this. 
This class is test class for VSTS task lib. Overall this is mock and test result.


```

I thought two possibility. 

1. make it constant 
   -> choose this.
2. make it constant with all other test mocks 
   -> it impact for the readability of the mock test code. 


```typescript 
let a: ma.TaskLibAnswers = <ma.TaskLibAnswers> {
    "which": {
        "echo":  "/usr/bin/echo"
    },
    "checkPath": {
        "./Tests/my-nginx.yml": true,
        "./Tests/kubectl": true,
        "./kubeconfig": true
    },
    "cwd": {
        "cwd": process.cwd(),
    },
    "osType": {
        "osType": "Linux",
    },
    "exec": {
       "./Tests/kubectl apply -f ./Tests/my-nginx.yml --kubeconfig ./kubeconfig": {
          "code": 0,
          "stdout": "deployment \"nginx-deployment\" created"  
       },
       "mkdir -p /opt/vsts/work/r1/a/.vstsbin": {
        "code": 0,
        "stdout": ""
      },
   } 
}
```

This part represent the mock and response. Also it is used for validation on the test class. The problem is , this part is JSON.  Which means if  the target value `"mkdir -p /opt/vsts/work/r1/a/.vstsbin"`  is coming, it returns `"code": 0,` and `"stdiout": ""`  Since this is json, I can't write something like this. 


```
      "mkdir -p " + DEFAULT_WORKING_DIRECTORY : {
             "code": 0, 
             "stdout": ""
       }
```

I can do it if I create a string then parse into the JSON object. However, following the coding style of the vsts task lib, it might not good. Since this is the mock value, and specify the value as the mock expectation, I recommend to leave this. 

However, if you have a good idea to refactor it and make it simple and DRY, I'd happy to learn and do it.  What do you say?